### PR TITLE
Add the situation of specifying spring.application.name in the bootstrap file after introducing spring cloud

### PR DESCRIPTION
**Pull request by AlchemyDing**
Add the situation of specifying spring.application.name in the bootstrap file after introducing spring cloud

### CODE REVIEW

File: instrumentation/spring/spring-boot-resources/library/src/main/java/io/opentelemetry/instrumentation/spring/resources/SpringBootServiceNameDetector.java [hunk 1]
```diff
@@ -84,7 +85,10 @@ public Resource createResource(ConfigProperties config) {
             this::findByCurrentDirectoryApplicationYaml,
             this::findByClasspathApplicationProperties,
             this::findByClasspathApplicationYml,
-            this::findByClasspathApplicationYaml);
+            this::findByClasspathApplicationYaml,
+            this::findByClasspathBootstrapProperties,
```
Comment (laurit) [inline instrumentation/spring/spring-boot-resources/library/src/main/java/io/opentelemetry/instrumentation/spring/resources/SpringBootServiceNameDetector.java:89]: @AlchemyDing did you verify that these have lower precedence than the source that are listed before them?

Comment (AlchemyDing) [reply instrumentation/spring/spring-boot-resources/library/src/main/java/io/opentelemetry/instrumentation/spring/resources/SpringBootServiceNameDetector.java:89]: Yes, when spring.application.name appears in both application.yml and bootstrap.yml, spring will take it from application.yml.

Comment (AlchemyDing) [reply instrumentation/spring/spring-boot-resources/library/src/main/java/io/opentelemetry/instrumentation/spring/resources/SpringBootServiceNameDetector.java:89]: in application.properties,spring.application.name=otel-demo-application;
in bootstrap.properties,spring.application.name=otel-demo-bootstrap-prop
environment.getProperty("spring.application.name")==otel-demo-application
image

Comment (AlchemyDing) [reply instrumentation/spring/spring-boot-resources/library/src/main/java/io/opentelemetry/instrumentation/spring/resources/SpringBootServiceNameDetector.java:89]: [https://docs.spring.io/spring-cloud-commons/docs/4.0.4/reference/html/#application-context-hierarchies](url)

Comment (AlchemyDing) [reply instrumentation/spring/spring-boot-resources/library/src/main/java/io/opentelemetry/instrumentation/spring/resources/SpringBootServiceNameDetector.java:89]: @laurit 
[https://docs.spring.io/spring-cloud-commons/docs/4.0.4/reference/html/#application-context-hierarchies](url)

Comment (AlchemyDing) [reply instrumentation/spring/spring-boot-resources/library/src/main/java/io/opentelemetry/instrumentation/spring/resources/SpringBootServiceNameDetector.java:89]: [https://docs.spring.io/spring-cloud-commons/docs/4.0.4/reference/html/#application-context-hierarchies](https://docs.spring.io/spring-cloud-commons/docs/4.0.4/reference/html/#application-context-hierarchies)
 @laurit

Comment (laurit) [reply instrumentation/spring/spring-boot-resources/library/src/main/java/io/opentelemetry/instrumentation/spring/resources/SpringBootServiceNameDetector.java:89]: Thanks for verifying this

**Review by laurit — APPROVED** [review]

File: instrumentation/spring/spring-boot-resources/library/src/main/java/io/opentelemetry/instrumentation/spring/resources/SpringBootServiceNameDetector.java [hunk 2]
```diff
@@ -262,6 +284,12 @@ private String readNameFromAppProperties() {
         "application.properties", SpringBootServiceNameDetector::getAppNamePropertyFromStream);
   }
 
+  @Nullable
+  private String readNameFromBootstrapProperties() {
```
Comment (mateuszrzeszutek) [inline instrumentation/spring/spring-boot-resources/library/src/main/java/io/opentelemetry/instrumentation/spring/resources/SpringBootServiceNameDetector.java:288]: Nit: method could be inlined

File: instrumentation/spring/spring-boot-resources/library/src/test/resources/bootstrap-multi.yml
Comment (mateuszrzeszutek) [inline instrumentation/spring/spring-boot-resources/library/src/test/resources/bootstrap-multi.yml:1]: This file does not actually contain multiple documents, it's no different from the regular `bootstrap.yaml` file

File: instrumentation/spring/spring-boot-resources/library/src/test/resources/bootstrap-multi.yml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+spring:
+  application:
+    name: cat-store-bootstrap
+
```
Comment (mateuszrzeszutek) [inline instrumentation/spring/spring-boot-resources/library/src/test/resources/bootstrap-multi.yml:4]: You need to use the `---` separator to have separate documents in a single file
```suggestion

---
```

Comment (AlchemyDing) [reply instrumentation/spring/spring-boot-resources/library/src/test/resources/bootstrap-multi.yml:4]: Fixed, thank you very much

**Review by mateuszrzeszutek — APPROVED** [review]
LGTM, thanks!

**Pull request merged**